File: Q22/Q22.py
#Pattern to be Made :
# if  N_Pattern = 5
#  
#  print ->
#           0 1 2 3 4 
#           1 2 3 4 5
#           2 3 4 5 6
#           3 4 5 6 7
#           4 5 6 7 8


def printPattern(N_pattern):
    start_ind = 0
    end_ind = N_pattern
    
    while start_ind<N_pattern:
        for printInd in range(start_ind,end_ind):
            print (printInd, end=' ')
        print('\r')
        start_ind = start_ind + 1
        end_ind = end_ind + 1








if __name__ == '__main__':
    N_pattern = int(input("Please enter the number of rows :"))

    printPattern(N_pattern)

File: Q22/Q22.java
public class Q22 {
    public static void main(String[] args) {
        var scanner = new Scanner(System.in);
        int numberOfRows;

        // the loop is used to check if the given number can make a pattern
        while (true) {
            System.out.print("Enter a number: ");
            numberOfRows = scanner.nextInt();
            if (numberOfRows > 0)
                break;
            else
                System.out.println("Invalid number. Please try again.");
        }
        printPattern(numberOfRows);
    }
    
    public static void printPattern(int nPattern){
        int startIdx = 0;
        int endIdx = nPattern;
      //Loop to print n differnet rows 
        while(startIdx < nPattern){
            //Loop to print n different cols
            for(int i = startIdx; i < endIdx; i++){
                System.out.print(i+" ");
            }
            System.out.println();
            startIdx++;
            endIdx++;
        }
    }
}
